cores/esp8266/core_esp8266_postmortem.cpp

```diff
@@ -128,6 +151,67 @@ void __wrap_system_restart_local() {
 
     ets_install_putc1(&uart_write_char_d);
 
+    /*
+      Check for Exceptions mapped into User Exception.
+    */
+    if (rst_info.reason == REASON_EXCEPTION_RST && rst_info.exccause == 0) {
+#if defined(DEBUG_ESP_PORT) || defined(DEBUG_ESP_EXCEPTIONS)
+        if ((rst_info.epc1 - 11u) == (uint32_t)_DoubleExceptionVector) {
+            struct EXC_CONTEXT *exc_context;
+            asm volatile("rsr.excsave2 %0\n\t" : "=a" (exc_context) :: "memory");
+            // Use Special Register values from before the `ill` instruction.
+            // Note, for a double exception the exccause is that of the second exception.
+            // epc1 is still the address of the 1st exception, its cause is now unknown.
+            rst_info.exccause = exc_context->exccause;
+            rst_info.epc1 = exc_context->epc1;
+            // Double Exception
+            if (rst_info.depc) {
+                ets_printf_P(PSTR("\nDouble Exception"));
+                // The current "ESP Exception Decoder" does not process the value in depc
+                // Set a fake epc1 so the decoder identifies the line of the Double Exception
+                rst_info.epc1 = rst_info.depc;
+                // The "ESP Exception Decoder" is also helped by updating the
+                // context saved on the stack.
+                exc_context->epc1 = rst_info.depc;
+            }
+            // BP - Debug Exception
+            else if (rst_info.epc2) {
+                if (rst_info.epc2 == 0x4000dc4bu) {
```

Comment:
What is this magic address? What is there?

Reply:
That is the address in the Boot ROM for the breakpoint instruction that executes for an unhandled exception. I'll replace with a `constexpr`.